ui/ducks/send/send.js

```diff
@@ -1530,31 +1552,38 @@ export function updateSendAsset({ type, details }) {
  * passing in both the dispatch method and the payload to dispatch, which makes
  * it only applicable for use within action creators.
  */
-const debouncedValidateRecipientUserInput = debounce((dispatch, payload) => {
-  dispatch(validateRecipientUserInput(payload));
+const debouncedValidateRecipientAddress = debounce((dispatch, payload) => {
+  dispatch(validateRecipientAddress(payload));
 }, 300);
 
 /**
  * This method is called to update the user's input into the ENS input field.
  * Once the field is updated, the field will be validated using a debounced
- * version of the validateRecipientUserInput action. This way validation only
+ * version of the validateRecipientAddress action. This way validation only
  * occurs once the user has stopped typing.
  *
  * @param {string} userInput - the value that the user is typing into the field
  */
 export function updateRecipientUserInput(userInput) {
   return async (dispatch, getState) => {
+    dispatch(actions.updateRecipientWarning('loading'));
     await dispatch(actions.updateRecipientUserInput(userInput));
     const state = getState();
     const chainId = getCurrentChainId(state);
     const tokens = getTokens(state);
     const useTokenDetection = getUseTokenDetection(state);
     const tokenAddressList = Object.keys(getTokenList(state));
-    debouncedValidateRecipientUserInput(dispatch, {
+    const { symbol, decimals } = await getSymbolAndDecimals(
+      userInput,
+      tokenAddressList,
+    );
```

Comment:
We might consider using `[getTokenStandardAndDetails](https://github.com/MetaMask/controllers/blob/a799f42ad7448a951ae02fd00be7fbfe6904aa8e/src/assets/AssetsContractController.ts#L167)` since it actually returns the contract standard and a more details with which we can inform the user more clearly about the contract they are sending to.

Reply:
☝️  This would also allow us to tell users if they are sending to an ERC721 or ERC1155 contract, which this current approach doesn't enable.